pulser-pasqal/pulser_pasqal/pasqal_cloud.py

```diff
@@ -159,15 +164,28 @@ def submit(
             emulator=emulator,
             strict_validation=mimic_qpu,
         )
-        create_batch_fn = backoff_decorator(self._sdk_connection.create_batch)
-        batch = create_batch_fn(
-            serialized_sequence=sequence.to_abstract_repr(),
-            jobs=job_params or [],  # type: ignore[arg-type]
-            emulator=emulator,
-            configuration=configuration,
-            wait=wait,
-        )
 
+        # If batch_id is not empty, thedn we can submit new jobs to a
+        # batch we just created otherwise, create a new one with
+        #  _sdk_connection.create_batch()
+        if batch_id:
```

Comment:
So this condition exists here because it's the only thing called via `run()` but if we expand run into another method for `append` jobs or something, we can dump the `if` and just have another method that calls `sdk_connection.add_jobs()`